File: internal/model/Printer.go
// Package model 定义应用核心数据类型:打印机、任务、设置。
package model

import "fmt"

// Conn 是打印机连接方式。
type Conn string

const (
	ConnNetwork Conn = "network" // 网口 IP:port
	ConnUSB     Conn = "usb"     // USB(经 Windows 打印驱动)
)

// Brand 是打印机品牌。佳博与飞蛾走同一套 ESC/POS 协议;其他也按 ESC/POS 处理。
// 品牌当前仅作元数据 + 未来品牌差异适配预留。
type Brand string

const (
	BrandGprinter Brand = "佳博"
	BrandFeie     Brand = "飞蛾"
	BrandOther    Brand = "其他"
)

// Brands 是下拉可选品牌顺序。
var Brands = []Brand{BrandGprinter, BrandFeie, BrandOther}

// BrandForIndex 按下拉索引返回品牌,越界回退佳博。
func BrandForIndex(i int) Brand {
	if i < 0 || i >= len(Brands) {
		return BrandGprinter
	}
	return Brands[i]
}

// IndexOfBrand 返回品牌在下拉中的索引,未匹配回退 0。
func IndexOfBrand(b Brand) int {
	for i, x := range Brands {
		if x == b {
			return i
		}
	}
	return 0
}

// Printer 是一台已注册的打印机。
type Printer struct {
	ID            string `json:"id"`
	Name          string `json:"name"`
	Brand         Brand  `json:"brand"`
	Width         int    `json:"width"` // 58 或 80
	Conn          Conn   `json:"conn"`
	IP            string `json:"ip"`      // 网口地址
	Port          string `json:"port"`    // 网口端口,默认 9100
	USBName       string `json:"usbName"` // USB:绑定的 Windows 打印机驱动名
	BuzzerEnabled bool   `json:"buzzer"`    // 打印时是否蜂鸣提示
	HeadLines     int    `json:"headLines"`   // 内容前空行数(走纸)
	TailLines     int    `json:"tailLines"`   // 尾部空行相对基数的偏移(可负;实际=escpos.TailFeed)
	CutDisabled   bool   `json:"cutDisabled"` // true=不切纸(该机无切刀);反向字段,默认 false=启用切刀
	LastPrint     string `json:"lastPrint"`
}

// Cuts 返回是否在打印末尾切纸(反向字段:零值即启用切刀,缺省切纸)。
func (p *Printer) Cuts() bool { return !p.CutDisabled }

// BrandLabel 返回品牌展示文本,空值回退「其他」。
func (p *Printer) BrandLabel() string {
	if p.Brand == "" {
		return string(BrandOther)
	}
	return string(p.Brand)
}

// ConnLabel 返回连接方式的中文标签。
func (p *Printer) ConnLabel() string {
	if p.Conn == ConnUSB {
		return "USB"
	}
	return "网口"
}

// Address 返回「地址 / 设备」列的展示文本。
func (p *Printer) Address() string {
	if p.Conn == ConnUSB {
		if p.USBName != "" {
			return p.USBName
		}
		return "USB 直连"
	}
	return p.IP + ":" + p.Port
}

// WidthLabel 返回规格标签,如 "80mm"。
func (p *Printer) WidthLabel() string {
	return fmt.Sprintf("%dmm", p.Width)
}

// Target 返回打印目标的展示文本,用于日志排查。
func (p *Printer) Target() string {
	if p.Conn == ConnUSB {
		return "USB『" + p.USBName + "』"
	}
	port := p.Port
	if port == "" {
		port = "9100"
	}
	return p.IP + ":" + port
}

File: Android/app/src/main/java/com/congmingpay/android/model/Printer.kt
package com.congmingpay.android.model

/**
 * 打印机连接方式。
 */
object Conn {
    const val NETWORK = "network"
    const val USB = "usb"
    const val BLUETOOTH = "bluetooth"
}

/**
 * 打印机品牌。佳博与飞蛾走同一套 ESC/POS 协议；其他也按 ESC/POS 处理。
 * 品牌当前仅作元数据 + 未来品牌差异适配预留。
 */
object Brand {
    const val GPRINTER = "佳博"
    const val FEIE = "飞蛾"
    const val OTHER = "其他"

    /** 下拉可选品牌顺序 */
    val ALL = listOf(GPRINTER, FEIE, OTHER)

    fun forIndex(i: Int): String =
        if (i in ALL.indices) ALL[i] else GPRINTER

    fun indexOf(brand: String): Int {
        val idx = ALL.indexOf(brand)
        return if (idx >= 0) idx else 0
    }
}

/**
 * 打印机登记来源（仅两种）。
 */
object Source {
    const val LOCAL = "local"
    const val CLOUD = "cloud"
}

/** 「上次打印时间」的固定墙钟格式 */
const val LAST_PRINT_TIME_LAYOUT = "yyyy-MM-dd HH:mm:ss"

/**
 * 打印机数据模型（可 Gson 序列化到 config.json）。
 */
data class Printer(
    var id: String = "",
    var name: String = "",
    var brand: String = Brand.OTHER,
    var width: Int = 80,
    var conn: String = Conn.NETWORK,
    var ip: String = "",
    var port: String = "",
    var usbName: String = "",
    var bluetoothMac: String = "",
    var buzzer: Boolean = true,
    var headLines: Int = 0,
    var tailLines: Int = 0,
    var cutDisabled: Boolean = false,
    var source: String = Source.LOCAL,
    var lastPrint: String = ""
) {
    /** 规范化来源（空=local） */
    fun effectiveSource(): String =
        if (source == Source.CLOUD) Source.CLOUD else Source.LOCAL

    fun sourceLabel(): String =
        if (effectiveSource() == Source.CLOUD) "云端下发" else "本地添加"

    /** 返回是否在打印末尾切纸（反向字段：默认 false=启用切刀） */
    fun cuts(): Boolean = !cutDisabled

    fun brandLabel(): String = if (brand.isEmpty()) Brand.OTHER else brand

    fun connLabel(): String = when (conn) {
        Conn.USB -> "USB"
        Conn.BLUETOOTH -> "蓝牙"
        else -> "网口"
    }

    fun address(): String = when (conn) {
        Conn.USB -> if (usbName.isNotEmpty()) usbName else "USB 直连"
        Conn.BLUETOOTH -> if (bluetoothMac.isNotEmpty()) bluetoothMac else "蓝牙"
        else -> "$ip:${if (port.isNotEmpty()) port else "9100"}"
    }

    fun widthLabel(): String = "${width}mm"

    fun target(): String = when (conn) {
        Conn.USB -> "USB『$usbName』"
        Conn.BLUETOOTH -> "蓝牙[$bluetoothMac]"
        else -> "$ip:${if (port.isNotEmpty()) port else "9100"}"
    }
}
